table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,389 @@
+      PathTranslator pathTranslator = mountAlluxioPaths(table, partitions);
+
+      // Glue does not provide column statistic information
+      List<ColumnStatisticsInfo> columnStatisticsData = new ArrayList<>();
+
+      PartitionInfo partitionInfo = PartitionInfo.newBuilder()
+          .setDbName(mGlueDbName)
+          .setTableName(tableName)
+          .addAllDataCols(GlueUtils.toProto(table.getStorageDescriptor().getColumns()))
+          .setStorage(GlueUtils.toProto(table.getStorageDescriptor(), pathTranslator))
+          .putAllParameters(table.getParameters())
+          .build();
+
+      Layout layout = Layout.newBuilder()
+          .setLayoutType(HiveLayout.TYPE)
+          .setLayoutData(partitionInfo.toByteString())
+          .build();
+
+      List<Column> partitionColumns = table.getPartitionKeys();
+
+      List<UdbPartition> udbPartitions = new ArrayList<>();
+      if (partitionColumns.isEmpty()) {
+        PartitionInfo.Builder partitionInfoBuilder = PartitionInfo.newBuilder()
+            .setDbName(mUdbContext.getDbName())
+            .setTableName(tableName)
+            .addAllDataCols(GlueUtils.toProto(table.getStorageDescriptor().getColumns()))
+            .setStorage(GlueUtils.toProto(table.getStorageDescriptor(), pathTranslator))
+            .setPartitionName(tableName)
+            .putAllParameters(table.getParameters());
+        udbPartitions.add(new GluePartition(
+            new HiveLayout(partitionInfoBuilder.build(), Collections.emptyList())));
+      } else {
+        for (Partition partition : partitions) {
+          String partName = GlueUtils.makePartitionName(partitionColumns, partition.getValues());
+          PartitionInfo.Builder pib = PartitionInfo.newBuilder()
+              .setDbName(getUdbContext().getDbName())
+              .setTableName(tableName)
+              .addAllDataCols(GlueUtils.toProto(partition.getStorageDescriptor().getColumns()))
+              .setStorage(GlueUtils.toProto(partition.getStorageDescriptor(), pathTranslator))
+              .setPartitionName(partName)
+              .putAllParameters(partition.getParameters());
+          if (partition.getValues() != null) {
+            pib.addAllValues(partition.getValues());
+          }
+          udbPartitions.add(new GluePartition(new HiveLayout(pib.build(),
+              statsMap.getOrDefault(partName, Collections.emptyList()))));
+        }
+      }
+
+      return new GlueTable(this,
+          pathTranslator,
+          tableName,
+          GlueUtils.toProtoSchema(table.getStorageDescriptor().getColumns()),
+          columnStatisticsData,
+          // Glue does not provide FieldSchema from API directly
+          // Get FieldSchema from partition keys
+          GlueUtils.toProto(table.getPartitionKeys()),
+          udbPartitions,
+          layout,
+          table);
+    } catch (EntityNotFoundException e) {
+      throw new NotFoundException("Table " + tableName + " does not exist.", e);
```

Comment:
For this exception and others in this method, can you add catalog id and database name?

Reply:
Added.

Reply:
I don't see the updated exception messages which include the db name and catalog id.